contractcourt/channel_arbitrator.go

```diff
@@ -1348,20 +1357,65 @@ func (c *ChannelArbitrator) channelAttendant(bestHeight int32,
 				}),
 			)
 
-			// We've cooperatively closed the channel, so we're no longer
-			// needed.
+		// We've cooperatively closed the channel, so we're no longer
+		// needed.
 		case <-c.cfg.ChainEvents.CooperativeClosure:
 			log.Infof("ChannelArbitrator(%v) closing due to co-op "+
 				"closure", c.cfg.ChanPoint)
 			return
 
+		// We have broadcasted our commitment, and it is now confirmed
+		// on-chain.
+		case closeSummary := <-c.cfg.ChainEvents.LocalUnilateralClosure:
```

Comment:
Would need to double check, but I think it's the case that for all of these events, we need to ensure we always pass in the _block height_ of the event. Otherwise, we may end up writing the incorrect height hint to disk, which has ramifications for neutrino (possibly missing an event).

Reply:
Now using the `spendingHeight` when advancing state. Also changed the variable name to `triggerHeight` to make it more clear.